src/wallet/bitcoin.ts

```diff
@@ -170,6 +168,23 @@ export class InMemoryBitcoinWallet implements BitcoinWallet {
     return "150";
   }
 
+  public async close(): Promise<any> {
+    const tasks = [];
+    if (this.pool.opened) {
+      tasks.push(await this.pool.close());
```

Comment:
I don't think that is what you want. You want to collect all the pending promises and `await` them all together at the end. 
I.e. 
```suggestion
      tasks.push(this.pool.close());
```

Reply:
You're totally right. Exactly what I wanted. Bloody JS.